### DISCUSSION TITLE
Nested groups

### DISCUSSION THREAD

**Discussion by marcuslindfeldt** [tailwindlabs/tailwindcss#13081] [Help]
Most web frameworks like e.g. React is based around the concept of reusable components. It's not uncommon to for example have a Tabs component. And its also not uncommon to have nested Tabs. This becomes problematic with the current tailwind group concept as the the group selector is not specific enough so it bleeds into the child tabs even though it shouldn't.

How do people normally solve this? Named groups doesn't work because it's a reusable component and you cant create dynamic names.

Currently I'm doing madness like this `[&_[role=tablist]_[role=tab]_[data-tab-text]:hover]:bg-muted` but it quickly becomes lengthy and fragile. E.g.

```
[
  `[&_[role=tablist]]:px-2 [&_[role=tablist]]:pt-2`,
  `[&_[role=tablist]_[role=tab]]:rounded-t-sm [&_[role=tablist]_[role=tab]]:relative [&_[role=tablist]_[role=tab]]:py-0 [&_[role=tablist]_[role=tab]:hover]:bg-muted [&_[role=tablist]_[role=tab]]  :!border-t [&_[role=tablist]_[role=tab]]:!border-inherit [&_[role=tablist]_[role=tab]:first-child]:border-l [&_[role=tablist]_[role=tab]]:border-r`,
  `[&_[role=tablist]_[role=tab]]:!bg-white [&_[role=tablist]_[aria-selected=true]:focus-visible_[data-tab-text]]:underline [&_[role=tablist]_[role=tab]:hover_[data-tab-text]]:text-foreground`,
  `[&_[role=tablist]_[aria-selected=true]]:before:content-[''] [&_[role=tablist]_[aria-selected=true]]:before:absolute [&_[role=tablist]_[aria-selected=true]]:before:bottom-[-1px] [&_[role=tablist]_  [aria-selected=true]]:before:w-full [&_[role=tablist]_[aria-selected=true]]:before:left-0 [&_[role=tablist]_[aria-selected=true]]:before:h-[1px] [&_[role=tablist]_[aria-selected=true]]  :before:bg-white`,
]
```

**Comment by wongjn** [comment]
You could consider having the classes on the elements themselves, rather than some parent element you seem to be added the utility classes to.
